File: Neon-Java/src/com/islandstudio/neon/Stable/secondary/iCommand/CommandHandler.java
package com.islandstudio.neon.stable.secondary.iCommand;

import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.List;

public interface CommandHandler {
    /**
     * Set the command handler for the particular feature.
     *
     * @param commander The player who perform the command.
     * @param args The command arguments.
     */
    void setCommandHandler(Player commander, String[] args);

    /**
     * Set the tab completion for the particular command.
     *
     * @param commander The player who perfom the command.
     * @param args The command arguments.
     * @return A list of valid command arguments.
     */
    default List<String> tabCompletion(Player commander, String[] args) {
        return new ArrayList<>();
    }
}

File: Neon-Kotlin/src/main/com/islandstudio/neon/stable/primary/nCommand/CommandHandler.kt
package com.islandstudio.neon.stable.primary.nCommand

import org.bukkit.entity.Player

interface CommandHandler {
    /**
     * Set the command handler for the particular feature.
     *
     * @param commander The player who perform the command.
     * @param args The command arguments.
     */
    fun setCommandHandler(commander: Player, args: Array<out String>)

    /**
     * Set the tab completion for the particular command.
     *
     * @param commander The player who perform the command.
     * @param args The command arguments.
     * @return A list of valid command arguments.
     */
     fun tabCompletion(commander: Player, args: Array<out String>): MutableList<String> {
        return mutableListOf()
    }
}